### DISCUSSION TITLE
Is the purpose of this project to interpret and comprehensively analyze the content of PDF documents?

### DISCUSSION THREAD

**Discussion by Bruce337f** [Filimoa/open-parse#43] [Q&A]
### Description

When I installed and ran the code according to the example, I easily obtained the text content existing on the pdf. This is a very convenient project!

But what puzzles me is that the developer also provided sample code for openai. Does this mean that openai can be provided to generate summary conclusions for PDF content, or analyze the theme of the content?

**Comment by Bruce337f** [comment]
I would like to ask if the following projects can edit PDF in detail and specific document content, and what is their relationship with this project?
Dealing with PDF's:

pdfminer.six Fully open source.
Extracting Tables:

PyMuPDF has some table detection functionality. Please see their license.
Table Transformer is a deep learning approach.
unitable is another transformers based approach with state-of-the-art performance.

This is a good project!

**Comment by Filimoa** [comment]
The first step of a RAG pipeline is splitting up the document into "semantic" chunks (parts of the doc that are talking about the same thing. This library is primarily aimed at that use case - embedding are a nice way of doing this.

**Reply by Bruce337f** [reply]
Thanks for your reply, I will understand this code again!

To be honest, it feels very similar to AI customer service, extraction-summarization-delivery

**Comment by Bruce337f** [comment]
About Semantic Processing Example:
```
from openparse import processing, DocumentParser
semantic_pipeline = processing.SemanticIngestionPipeline(
    openai_api_key=OPEN_AI_KEY,
    model="text-embedding-3-large",
    min_tokens=64,
    max_tokens=1024,
)
parser = DocumentParser(
    processing_pipeline=semantic_pipeline,
)
parsed_content = parser.parse(basic_doc_path)
```
Could you please tell me if it can be combined with the code below，

```
from llama_index.core import VectorStoreIndex
index = VectorStoreIndex(nodes=nodes)
query_engine = index.as_query_engine()
response = query_engine.query("What do they do to make money?")
print(response)
```
Form a "answer every question" PDF analysis assistant?
